group/short.go

```diff
@@ -226,9 +228,20 @@ func (e *wElt) MarshalBinary() ([]byte, error) {
 	if e.IsIdentity() {
 		return []byte{0x0}, nil
 	}
+
+	ec := e.wG.c
+	if e.wG == P384 {
+		ec = elliptic.P384()
+	}
```

Comment:
Why this line?

Reply:
this is because the [`ECDH` function](https://cs.opensource.google/go/go/+/refs/tags/go1.24.5:src/crypto/ecdsa/ecdsa.go;l=52) checks whether ec is an `elliptic.Curve` object.

I made changes to avoid this check.